sheepdog/utils/transforms/__init__.py

```diff
@@ -204,7 +204,7 @@ def get_converted_type_from_list(cls, prop_name, value):
                 return parse_list_from_string(value)
             elif value_type == float:
                 if float(value).is_integer():
-                    return int(value)
+                    return int(float(value))
                 else:
                     return float(value)
```

Comment:
this logic doesn't make much sense to me, i'm not sure why we didn't just do `if value_type == float: return float(value)` instead of checking `is_integer` in the first place. can you try that and see if other tests break?

Reply:
I think this is done because some fields have type `integer` so if someone uses `2.0` instead of `2` it will still work. We could change it to just include another elif for integer types.

Reply:
oh, i just realized now that there's no `if` for int because of the dictionary types! it makes sense then